Restoring backward compatibility, refactoring

oleduc opened the pull request:
This pull requests is a response to perceived source decay. Its mostly de-duplication and cleanup but I also had to revert a few things @vkosuri did.

Here is a breakdown of what I've done and why:

**1 - Removed retry decorator**
I remove it because of two reasons. Firstly because it was not very fast and it was implemented using a hard sleep. Also requests already supports adapter level max_retries. Secondly, the decorator would break Robotframework's parameter translation, which in turn made it impossible to use strings containing the equal symbol as it would be interpreted as a keyword argument.

**2- Removed overly broad catch around requests calls**
It would cause every exception to be interpreted as a retry exception. Also its not necessary, we want requests exceptions to reach the framework if possible.

**3- Deduplication**
A lost of the debug code was just copy pasted and needed to be refactored.

**4- Made arguments optional again**
Because it broke backward compatibility.

Also @bulkan if you want help with code reviews and maintenance, me and my employers would gladly help around if you provided us with collaborator access :)

Closes #84 
Closes #83
Includes #85

vkosuri commented:
Few more comments

If you are not using requests exception, it doesn't required

``` LINE 12: from requests.exceptions import RequestException```

timeout should in float, because robot data is in unicode format (not always)

``` LINE 78: self.timeout = float(timeout) ```

LINE 175 and 177 does it require?

```  `max_delay` The maximum number of delay each connection should attempt.
        
        `max_backoff` Expoential backoff
```

Does debug messages are printing into robot log?

oleduc commented:
1 - For the max_retry, I'm not sure what is the use case for more granular control over max_retry for the same session.
2 - I remove unused impors.
3 - Good point, I corrected that.
**LINE 175 and 177 does it require?**
No I removed it.
**Does debug messages are printing into robot log?**
Yes, I didn't change the mechanism you implemented, I just extracted it into a function.

oleduc commented:
@vkosuri Does this look good to you?

vkosuri commented:
@oleduc thank you nice work, 

@bulkan any comments from your side?

@oleduc if possible could you also add some testcases for url params for post, patch and delete after this merge? and documentation generation?

oleduc commented:
@vkosuri Yes, good idea.

vkosuri commented:
@oleduc working on this issue https://github.com/bulkan/robotframework-requests/issues/83

vkosuri commented:
@oleduc few more comments on this PR

  Instead print messages for deprecation warning i think it is beter to having logger.warn

``` Python 
print "Deprication Warning  Use Get Request in the future"
```
``` Python 
logger.warn("Deprication Warning  Use Get Request in the future")
```

  Enable reStructed for doc strings for all deprecated doc string

```* * *   Deprecated- See Post Request now   * * *```

```**Deprecated- See Post Request now**```

  We are using last_resp for history, i think this is not required

```(@Kosuri Why?)```

  doc string position changes, even though we are using named arguments, i think it is better to having in proper position

``` LINE 349: `params` url parameters to append to the uri````


  def post,  patch and delete doesn't having params, could you please add params arguments these procedures also?

``` Python
def post(self, alias, uri, data={}, headers=None, files=None, allow_redirects=None, timeout=None):
 def patch(self, alias, uri, data={}, headers=None, files={}, allow_redirects=None, timeout=None):
def put(self, alias, uri, data=None, headers=None, allow_redirects=None, timeout=None):
def delete(self, alias, uri, data=(), headers=None, allow_redirects=None, timeout=None):
```

vkosuri commented:
https://github.com/bulkan/robotframework-requests/pull/81

vkosuri commented:
@oleduc any updates on this?

@bulkan any comments on this PR?

bulkan commented:
I just came back from holidays will look at this in a day or two

vkosuri commented:
@bulkan any comments on this PR?

bulkan commented:
@vkosuri once @oleduc updates PR with your last suggestion it looks good to merge

oleduc commented:
@vkosuri ***def post, patch and delete doesn't having params, could you please add params arguments these procedures also?***

These methods are deprecated, we should keep their API as is.

vkosuri commented:
@oleduc thank you agreed with you

@bulkan it looks good to me, could merge this PR?

bulkan commented on src/RequestsLibrary/RequestsKeywords.py:
```diff
@@ -325,20 +253,18 @@ def get(self, alias, uri, headers=None, params={}, allow_redirects=None, timeout
         `timeout` connection timeout
-        """        
+        """
         print "Deprication Warning  Use Get Request in the future"
```
this is still here, can we change to logger.warn ?

oleduc commented:
@bulkan Sorry I didn't have time to do it today. I'll try to have it done by 03/02/2016 @ 23:00 GMT-5

vkosuri commented:
@oleduc any updates?

bulkan commented:
@vkosuri if i merge this can you fix up the `print` to `warn` ?

vkosuri commented:
@bulkan yeah sure

bulkan commented:
@vkosuri ok merged :)

oleduc commented:
@bulkan Sorry, I can do it really quickly.
